src/setup/node.rs

```diff
@@ -43,37 +43,69 @@ pub enum ChildExitCode {
     ErrorCode(Option<i32>),
 }
 
+/// Node type is used to select different startup configurations.
+pub enum NodeType {
+    /// A non-validator node without any preloaded data.
+    Stateless,
+    /// A validator node with a preloaded ledger data.
+    Stateful
+}
+
 pub struct NodeBuilder {
     conf: NodeConfig,
     meta: NodeMetaData,
 }
 
 impl NodeBuilder {
     /// Creates new [NodeBuilder]. Initial state is taken from `state` path, the node will run in `target` path.
-    pub fn new(state: Option<PathBuf>, target: PathBuf) -> Result<Self> {
-        if !target.exists() {
-            fs::create_dir_all(&target)?;
-        }
-        let mut copy_options = CopyOptions::new();
-        copy_options.content_only = true;
-        copy_options.overwrite = true;
+    pub fn new() -> Result<Self> {
         let setup_path = build_ripple_work_path()?.join(RIPPLE_SETUP_DIR);
-        copy(&setup_path, &target, &copy_options)?;
```

Comment:
In the legacy code, we copied the whole setup directory, which is not necessary.
TempDir doesn't care about the config.toml file.